__init__.py

```diff
@@ -129,25 +129,115 @@ def import_sensor_data(dryrun: bool = False):
 
-            # required by timely_beliefs, TODO: check if that still is the case, see https://github.com/SeitaBV/timely-beliefs/issues/64
-            df_sensor.index.name = "event_start"
-            df_sensor.name = "event_value"
+            save_new_beliefs(df_sensor, data_source, sensor, now)
 
-            bdf = BeliefsDataFrame(
-                df_sensor,
-                source=data_source,
-                sensor=sensor,
-                belief_time=now,
+        # Save derived sensors
+        sensors = ensure_zinfo_derived_sensors()
+        for sensor in sensors:
+            zinfo_sensor_name = sensor.zinfo_sensor_name
+            if isinstance(zinfo_sensor_name, list):
+                mask = df.index.get_level_values(zinfo_sensor_name_field).isin(
+                    zinfo_sensor_name
+                )
+            else:
+                mask = (
+                    df.index.get_level_values(zinfo_sensor_name_field)
+                    == zinfo_sensor_name
+                )
+            df_sensor = df.loc[mask]
+            df_sensor = apply_pandas_method_kwargs(
+                df_sensor, sensor.pandas_method_kwargs
             )
+            df_sensor = df_sensor[zinfo_event_value_field]
+
+            save_new_beliefs(df_sensor, data_source, sensor, now)
+
+
+def save_new_beliefs(df_sensor, data_source, sensor, belief_time) -> BeliefsDataFrame:
+    # required by timely_beliefs, TODO: check if that still is the case, see https://github.com/SeitaBV/timely-beliefs/issues/64
+    df_sensor.index.name = "event_start"
+    df_sensor.name = "event_value"
+
+    bdf = BeliefsDataFrame(
+        df_sensor,
+        source=data_source,
+        sensor=sensor,
+        belief_time=belief_time,
+    )
+
+    # Drop beliefs that haven't changed
+    bdf = drop_unchanged_beliefs(bdf)
+
+    # TODO: evaluate some traits of the data via FlexMeasures, see https://github.com/SeitaBV/flexmeasures-entsoe/issues/3
+    save_to_db(bdf)
 
-            # Drop beliefs that haven't changed
-            bdf = drop_unchanged_beliefs(bdf)
 
-            # TODO: evaluate some traits of the data via FlexMeasures, see https://github.com/SeitaBV/flexmeasures-entsoe/issues/3
-            save_to_db(bdf)
+def apply_pandas_method_kwargs(
+    df: pd.DataFrame, pandas_method_kwargs: List[Tuple[str, dict]]
+) -> pd.DataFrame:
+    """Convert the data frame using the given list.
+
+    The conversion is defined using `pandas_method_kwargs`,
+    which lists method/kwargs tuples that are called in the specified order.
+    The example below converts from hourly meter readings in kWh to electricity demand in kW.
+
+        pandas_method_kwargs=[
+            ("diff", dict()),
+            ("shift", dict(periods=-1)),
+            ("head", dict(n=-1)),
+        ],
+    """
+    for method, kwargs in pandas_method_kwargs:
+        df = getattr(df, method)(**kwargs)
+    return df
+
+
+def ensure_zinfo_derived_sensors() -> List[Sensor]:
```

Comment:
I don't see (wihtout too much effort) what the difference in fucntionality is between this function and `ensure_zinfo_main_sensors`. Please say so in the docstring, and/or maybe refactor a bit if useful.

Reply:
I now refactored the config setting to streamline the two different ways I had to loop through the main and derived sensor descriptions.